miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryDilutionService.java

```diff
@@ -57,7 +57,19 @@ public LibraryDilution get(long dilutionId) throws IOException {
     return dilution;
   }
 
+  private boolean isTargetedSequencingInLibrary(LibraryDilution dilution) {
+    return dilution.getLibrary().getKitDescriptor().getTargetedSequencing().contains(dilution.getTargetedSequencing());
+  }
+
+  private void validateTargetedSequencing(LibraryDilution dilution) {
+    if (!isTargetedSequencingInLibrary(dilution)) {
+      throw new IllegalArgumentException("Dilution Targeted Sequencing not contained in parent library");
```

Comment:
Since the issue is with the chosen kit descriptor and not with the library itself per se, what do you think about changing this to something like `Selected targeted sequencing is not available for kit on parent library` ?
For everyone: since this is a rare case (and more likely to happen with legacy dilutions as opposed to newly-created ones), do we want to append suggestions for how to fix it? Something like `If you think this is an error, please contact your MISO administrator to make targeted sequencing ${targetedSequencingName} available for kit ${kitName}.`

Reply:
@callunity Sounds good, I can implement these.

I'll wait for consensus on the 2nd suggestion, but I can do that as well.

Reply:
:+1:

There are roughly 3232 dilutions with incompatible targeted sequencing in OICR production right now. All of them are legacy/migrated from previous LIMS

```
SELECT d.dilutionId, d.preMigrationId, l.libraryId, l.alias
FROM LibraryDilution d
JOIN Library l ON l.libraryId = d.library_libraryId
JOIN DetailedLibrary dl ON dl.libraryId = l.libraryId
JOIN LibraryDesignCode ldc ON ldc.libraryDesignCodeId = dl.libraryDesignCodeId
WHERE ldc.targetedSequencingRequired = 1
AND d.targetedSequencingId IS NOT NULL
AND d.targetedSequencingId NOT IN (
  SELECT tk.targetedSequencingId FROM TargetedSequencing_KitDescriptor tk
  WHERE tk.kitDescriptorId = l.kitDescriptorId
);
```